connector/avro/src/main/scala/org/apache/spark/sql/avro/AvroDeserializer.scala

```diff
@@ -128,6 +142,26 @@ private[sql] class AvroDeserializer(
       case (INT, IntegerType) => (updater, ordinal, value) =>
         updater.setInt(ordinal, value.asInstanceOf[Int])
 
+      case (INT, DateType) if isNotExactType =>
```

Comment:
How about moving the check here as well as the code changes above to a new method?

Reply:
yeah, i can

Reply:
It may make the code looks weird. Just stay what it is now